From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 20 Aug 2026 00:18:51 +0200
Subject: popupMenu: Use connectObject to connect to click gestures signals
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

We do not want to potentially access to the callback if the menu is
destroyed as part of the gesture event.

  (gnome-shell:1879091): Gjs-CRITICAL **: 00:23:05.833: Object
  Gjs_gdm_authMenuButton_AuthMenuItem (0x5eb9d7335490), has been already
  disposed — impossible to access it. This might be caused by the object having
  been destroyed from C code using something such as destroy(), dispose(), or
  remove() vfuncs.
  == Stack trace for context 0x5eb9d350fe60 ==
  #0 i resource:///org/gnome/shell/ui/popupMenu.js:115
  #1 i resource:///org/gnome/shell/ui/init.js:20
---
 js/ui/popupMenu.js | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/js/ui/popupMenu.js b/js/ui/popupMenu.js
index 32eb357..8e4eb02 100644
--- a/js/ui/popupMenu.js
+++ b/js/ui/popupMenu.js
@@ -106,14 +106,14 @@ export const PopupBaseMenuItem = GObject.registerClass({
         this._clickGesture = new Clutter.ClickGesture({
             enabled: this._activatable,
         });
-        this._clickGesture.connect('recognize',
-            () => this.activate(Clutter.get_current_event()));
-        this._clickGesture.connect('notify::pressed', () => {
+        this._clickGesture.connectObject('recognize',
+            () => this.activate(Clutter.get_current_event()), this);
+        this._clickGesture.connectObject('notify::pressed', () => {
             if (this._clickGesture.pressed)
                 this.add_style_pseudo_class('active');
             else
                 this.remove_style_pseudo_class('active');
-        });
+        }, this);
         this.add_action(this._clickGesture);
 
         if (!this._activatable)
